homework-submission/week2/app.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+// This component renders form element
+const TodoForm = ({ children, name, style }) => (
+  <form name={name} className={style}>
+    {children}
+  </form>
+);
+
+// This component renders given types of input element
+const TodoInput = ({ type, name, placeholder, style, onChangeFunction }) => (
+  <input
+    className={style}
+    placeholder={placeholder}
+    onChange={onChangeFunction}
+    type={type}
+    name={name}
+  />
+);
+
+// This component renders button element
+const TodoButton = ({ onClickFunction, style, children }) => (
+  <button className={style} onClick={onClickFunction}>
+    {children}
+  </button>
+);
+
+// This component renders <li>  items into <ul> element(TodoList component)
+const Todo = props => {
+  const { todo, toggleDone, removeTodo } = props;
+  const { description, deadline, done } = todo;
+
+  return (
+    <li className="todo">
+      <span onClick={toggleDone} className={done ? 'done' : 'not-done'}>
+        {description},{deadline}
+      </span>
+      <TodoButton style="remove" onClickFunction={removeTodo}>
+        x
+      </TodoButton>
+    </li>
+  );
+};
+
+//This component renders ul element h2 title element for Todo list
+const TodoList = ({ name, style, children }) => (
+  <ul className={style}>
+    <h2>{name}</h2>
+    {children}
+  </ul>
+);
+
+//This is the main component of the Todo App
+class App extends React.Component {
+  state = {
+    dynamicList: [
+      {
+        description: 'Get out of bed',
+        deadline: '2017-09-11',
+        done: true,
+      },
+      {
+        description: 'Brush teeth',
+        deadline: '2017-09-10',
+        done: false,
+      },
+      {
+        description: 'Eat breakfast',
+        deadline: '2017-09-09',
+        done: false,
+      },
+    ],
+    formData: {
+      description: '',
+      deadline: '',
+      done: false,
+    },
+  };
+
+  toggleDone = index => {
+    this.setState(state => (state.dynamicList[index].done = !state.dynamicList[index].done));
```

Comment:
does this work?  You are passing a function that returns a boolean value to this.setState?